docs/source/notes/cuda.rst

```diff
@@ -54,6 +54,68 @@ Below you can find a small example showcasing this::
         f = torch.randn(2).cuda(cuda2)
         # d.device, e.device, and f.device are all device(type='cuda', index=2)
 
+.. _tf32_on_ampere:
+
+TensorFloat-32(TF32) on Ampere devices
+--------------------------------------
+
+Starting in PyTorch 1.7, there is a new flag called `allow_tf32` which defaults to true.
+This flag controls whether PyTorch is allowed to use the TensorFloat32 (TF32) tensor cores,
+available on new NVIDIA GPUs since Ampere, internally to compute GEMMs and convolutions.
+
+TF32 tensor cores are designed to achieve better performance on GEMMs and convolutions on
+`torch.float32` tensors by doing arithmetics at the precision of `torch.half` while still
+maintaining the dynamic range of `torch.float32`.
+
+GEMMs and convolutions are controlled separately, and their corresponding flag can be accessed at:
+
+.. code:: python
+
+  # The flag below controls whether to allow TF32 on GEMMs. This flag defaults to True.
+  torch.backends.cuda.matmul.allow_tf32 = True
+
+  # The allow_tf32 flag for convolutions is not implemented yet
+
+To get an idea of the precision and speed, see the example code below:
+
+.. code:: python
+
+  a_full = torch.randn(10240, 10240, dtype=torch.double, device='cuda')
+  b_full = torch.randn(10240, 10240, dtype=torch.double, device='cuda')
+  ab_full = a_full @ b_full
+
+  a = a_full.float()
+  b = b_full.float()
+
+  # Do matmul at TF32 mode.
+  ab_tf32 = a @ b  # takes 0.018s on GA100
+  error = (ab_tf32 - ab_full).abs().max()  # 0.1713
```

Comment:
Showing the absolute error is interesting but you may also want to show another statistic, like average relative error, since it's hard to understand how significant 0.1713 is. If the values are small then the error is massive, and if the values are very large then the error is likely inconsequential.

Reply:
The maximum relative error is ~80000, I believe this is because there is some numbers in ab_full being very close to 0. But I believe I can use
```
(ab_tf32 - ab_full).abs().max() / ab_full.abs().mean()
```

Reply:
fixed